tests/lean/1804.lean

```diff
@@ -0,0 +1,24 @@
+def inc : StateM Nat Nat := do
+  let s ← get
+  modify (· + 1)
+  return s
+
+def f (x : Bool) : StateM Nat Nat := do
+  let .true := x | return (← inc)
+  get
+
+def g (x : Bool) : StateM Nat Nat := do
+  let .true := x | do return (← inc)
+  get
+
+#eval g true |>.run' 0 -- `0` as expected
+#eval f true |>.run' 0 -- `1` :(
```

Comment:
Opinions: should I change this test to show the new (fixed) behavior or keep the commentary that describes a bug that no longer exists but is being protected against regressions?

Reply:
I usually change them to something like "should not return `1`"